### DISCUSSION TITLE
Better "contents" visualization

### DISCUSSION THREAD

**Discussion by AlessandroSpallina** [outline/outline#4307] [Ideas]
Hello,
I'm using outline a lot with my team and I love it, so thanks for this fantastic piece of sw!

I have a few ideas for a little improvement in the "contents" area:

* More tabulation, working with big pages with a lot of different kinds of titles (H1-H3) the visualization is not clear enough in my opinion, a bigger tabulation to distinguish the H1 title from the H2 title from the H3 title can be a fix.
  Follows the screenshot of the current contents area.
  ![outline-pre](https://user-images.githubusercontent.com/10786872/196213210-b7d6d2fa-c351-4f5e-92da-23bbcc9b30a2.png)
  Modified CSS to have a better tabulation (margin-left=1em)
  ![outline-post](https://user-images.githubusercontent.com/10786872/196213217-8d67a267-9198-47e5-8185-05be34b3d0d9.png)

* The contents area should support title collapse/expand, the default view should be expanded 
* The contents area should be resizable like the main menu
* Automatically add section/paragraph numbers? (e.g. "1 - H1 title", "1.1 - H2 title", "1.1.1 - H3 title")

**Comment by ReessKennedy** [comment]
I agree with you @AlessandroSpallina ... buy you could also modify the CSS yourself. I just went through this and wrote solution I found as follow up here: https://github.com/outline/outline/discussions/4206 ... but I understand/appreciate you're suggesting an improvement like this be made for everyone by the devs.

**Reply by EduNetEurope** [reply]
Sounds very interesting!!! Thanks for sharing!

**Comment by EduNetEurope** [comment]
@tommoor 
It would be amazing if we could add custom CSS that is not overwritten by updates.
It could take care of a number of feature requests:

- https://github.com/outline/outline/discussions/4206
- https://github.com/outline/outline/discussions/6384
- https://github.com/outline/outline/discussions/6746
- ...
